app/controllers/api/v1/reports_controller.rb

```diff
@@ -0,0 +1,38 @@
+class API::V1::ReportsController < API::APIController
+  rescue_from Pundit::NotAuthorizedError, with: :pundit_user_not_authorized
+
+  def pundit_user_not_authorized(exception)
+    unauthorized
+  end
+
+  # GET api/v1/reports/:id
+  def show
+    offering = Portal::Offering.find(params[:id])
+    # authorize offering, :api_report?
+    render json: API::V1::Report.new(offering, request.protocol, request.host_with_port).to_json
+  end
+
+  # PUT api/v1/reports/:id
+  def update
+    offering = Portal::Offering.find(params[:id])
+    # authorize offering, :api_report?
+    puts params
+    if params[:visibility_filter]
+      update_visibility_filter(offering.report_embeddable_filter, params[:visibility_filter])
+    end
+    offering.update_attributes!(report_params)
+    head :ok
+  end
+
+  private
+
+  def update_visibility_filter(filter, filter_params)
+    filter.embeddable_keys = filter_params[:questions] unless filter_params[:questions].nil?
```

Comment:
Do mean the explicit `.nil?` check might be confusing (as `present?` / `blank?` would fail here for empty array edge-case)?

Otherwise I'm not sure if I understand. Basically the filter consists of list of questions + enabled / disabled flag (named `ignore`). They can be set independent, so you can keep questions selected, but still deactivate the filter. And this API seems to reflect that, just sets two properties + makes sure we won't unset the second property if only one is provided.